app/Http/Controllers/CompetitionsController.php

```diff
@@ -94,15 +95,38 @@ public function destroy(Competition $competition)
     }
 
     /**
-     * Download the CSV export of all users.
+     * Download a CSV export of all users based on different criteria.
      *
      * @param  \Gladiator\Models\Competition  $competition
-     * @return \League\Csv\ $csv
+     * @param  \Illuminate\Http\Request  $request
+     * @return void
      */
-    public function export(Competition $competition)
+    public function export(Competition $competition, Request $request)
     {
-        $csv = $this->manager->exportCSV($competition);
-        $csv->output('competition' . $competition->id . '.csv');
+        $fileName = 'contest_' . $competition->contest_id . '-' . 'competition_' . $competition->id;
+
+        $withReportback = convert_string_to_boolean($request->input('withReportback'));
+
+        $users = $this->manager->getModelUsers($competition, $withReportback);
+
+        if (! $withReportback) {
+            // @TODO: can probably flash this to the session too!
+            $fileName = $fileName . '-users';
+        } else {
+            $fileName = $fileName . '-leaderboard';
+
+            if (session()->has($fileName)) {
```

Comment:
@deadlybutter I store the session here: https://github.com/DoSomething/gladiator/pull/224/files#diff-626b13c4d9542a855923a32d7ed83d74R177